src/KubernetesClient.Informers/ResourceObservableExtensions.cs

```diff
@@ -0,0 +1,273 @@
+                                .Select(x => x.Value) // stuff in buffer that also existed in cache (by key)
+                                .Except(cacheSnapshot.Select(x => x.Value), comparer)
+                                .Select(x => x.ToResourceEvent(EventTypeFlags.Modify | EventTypeFlags.Computed))
+                                .ToList();
+                        }
+
+                        var updatedKeys = updatedEntities
+                            .Select(x => keySelector(x.Value))
+                            .ToHashSet();
+
+                        var resetEntities = resetBuffer
+                            .Select(x => x.Value)
+                            .Where(x => x != null &&
+                                        !addedKeys.Contains(keySelector(x)) &&
+                                        !deletedKeys.Contains(keySelector(x)) &&
+                                        !updatedKeys.Contains(keySelector(x)))
+                            .ToReset()
+                            .ToList();
+
+                        return deletedEntities
+                            .Union(addedEntities)
+                            .Union(updatedEntities)
+                            .Union(resetEntities);
+                    })
+                    .SynchronizeCache(cache, keySelector)
+                    .Do(msg => { cacheSynchronized = true; })
+                    .Select(x => x.Value)
+                    .ObserveOn(Scheduler.Immediate)
+                    .Subscribe(observer);
+            });
+        }
+
+        /// <summary>
+        ///     Injects a <see cref="ResourceEvent{T}" /> of type <see cref="ResourceStreamType.Sync" /> into the observable for each item produced
+        ///     by the <see cref="ResourceStreamType.List" /> operation from <paramref name="source" />
+        /// </summary>
+        /// <param name="source">The source sequence that will have sync messages appended</param>
+        /// <param name="timeSpan">The timespan interval at which the messages should be produced</param>
+        /// <typeparam name="T">The type of resource</typeparam>
+        /// <returns>Original sequence with resync applied</returns>
+        public static IObservable<ResourceEvent<T>> Resync<T>(this IObservable<ResourceEvent<T>> source, TimeSpan timeSpan)
+        {
+            return Observable.Create<ResourceEvent<T>>(observer =>
+            {
+                var timerSubscription = Observable
+                    .Interval(timeSpan)
+                    .SelectMany(_ => source
+                        .TakeUntil(x => x.EventFlags.HasFlag(EventTypeFlags.ResetEnd))
+                        .Do(x =>
+                        {
+                            if (!x.EventFlags.HasFlag(EventTypeFlags.Reset))
+                                throw new InvalidOperationException("Resync was applied to an observable sequence that does not issue a valid List event block when subscribed to");
```

Comment:
Throwing in a `Do` block is probably Not Great, a better way would be to write:

```csharp
.SelectMany(x => x.EventFlags.HasFlag(EventTypeFlags.Reset) ? 
  Observable.Throw(new InvalidOperationException("...") :
  Observable.Return(x));
```

Reply:
That's cleaner but `Do` tends to leave the user with no doubt what's happening.

Judgement call.  I prefer Ana's way but for the less Rx-minded, `Do` is fine.

Reply:
Unless there is a specific issue with the way the code is currently written, I think I'll leave at as `.Do`. I feel like the proposed method will actually be significantly heavier as it would allocate a new observable for EACH message.